control-plane/pstor-proxy/proto/v1/frontend_node/registration.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+import "v1/misc/common.proto";
+
+package pstor_proxy.v1;
+
+// Register the frontend node with control-plane
+message RegisterRequest {
+  // Id of the frontend node instance
+  string id = 1;
+  // Grpc endpoint of the frontend node instance
+  string grpc_endpoint = 2;
+  // Frontend node labels
+  optional StringMapValue labels = 3;
+}
```

Comment:
How about importing the message type FrontendNodeSpec here and do:
```suggestion
message RegisterRequest {
  FrontendNodeSpec frontend_node = 1;
}
```

Reply:
Yes we can, but in future if we have more fields in frontend node spec, which might not be a part of registration.